src/robot/running/userkeywordrunner.py

```diff
@@ -34,6 +34,7 @@ class UserKeywordRunner(object):
     def __init__(self, handler, name=None):
         self._handler = handler
         self.name = name or handler.name
+        self.sourcename = handler.name
```

Comment:
There shouldn't be need to assign `self.sourcename` when we can access it via `self._handler`. Exactly like with library keywords, it ought to be possible to enhance only the `EmbeddedArgumentsRunner`.